ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Config.hs

```diff
@@ -28,23 +31,36 @@ import           Ouroboros.Consensus.Shelley.Ledger.Block
   Additional node configuration
 -------------------------------------------------------------------------------}
 
+-- | In case we're a block issuer, the verification key we use.
+data BlockIssuerVKey c =
+    BlockIssuerVKey !(SL.VKey 'SL.BlockIssuer c)
+  | NotABlockIssuer
+  deriving stock (Show, Generic)
+  deriving anyclass (NoUnexpectedThunks)
+
 data instance BlockConfig (ShelleyBlock c) = ShelleyConfig {
       -- | The highest protocol version this node supports. It will be stored
       -- the headers of produced blocks.
       shelleyProtocolVersion :: !SL.ProtVer
     , shelleySystemStart     :: !SystemStart
     , shelleyNetworkMagic    :: !NetworkMagic
     , shelleyProtocolMagicId :: !ProtocolMagicId
+    , shelleyBlockIssuerVKey :: !(BlockIssuerVKey c)
```

Comment:
Comment: When chain selection is comparing two fragments, it will prefer the fragment with a tip signed by this key (provided that the `BlockNo`s and `SlotNo`s of the two tips are equal). For core nodes, this should be set to the verification key corresponding to the node's signing key, to make sure we prefer self-issued blocks. For non core codes, this can be set to `NotABlockIssuer`.